### PR TITLE
Ciac 7465 & 7466/ Add Delete Existing Data Set Flag & Re-Generate Event Id

### PR DESCRIPTION
**Pull request by AradCarmi**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-7465
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-7466

## Description
- Remove the event ID from the dataset.
- Create a new event ID.
- Verify the new event ID is unique.
- Added the -dd flag to the **modeling-rules** *test* command when enabled it is deleting the dataset if it exists on the tenant.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Changelog
 ## Unreleased
+* Added the *--delete_existing_dataset/-dd* flag to the **modeling-rules test** command to delete an existing dataset in the tenant.
```
Comment (daryakoval) [inline CHANGELOG.md:3]: What about - verify_event_id_does_not_exist_on_tenant ?

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 2]
```diff
@@ -602,6 +607,127 @@ def verify_test_data_exists(test_data_path: Path) -> Tuple[List[str], List[str]]
...
+            logger.info(
+                "[green]The event ids does not exists on the tenant[/green]",
+                extra={"markup": True},
+            )
+            return
+    logger.error(
+        ("[red]The event id already exists in the tenant[/red]"),
+        extra={"markup": True},
+    )
+    raise typer.Exit(1)
```
Comment (daryakoval) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:648]: Why fail the test on it? 
In the previous PR this was not like this?
Where do you replace the existing uuid with new one?

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 3]
```diff
@@ -602,6 +607,127 @@ def verify_test_data_exists(test_data_path: Path) -> Tuple[List[str], List[str]]
...
+        timeout (int): The timeout in seconds.
+        interval (int): The interval in seconds.
+        ignore_errors (bool): Ignore errors.
+    """
+    logger.info(
+        f'[cyan]Checking if dataset "{dataset}" exists on the tenant...[/cyan]',
+        extra={"markup": True},
+    )
+    query = f"config timeframe = 10y | dataset = {dataset}"
+    for i in range(timeout // interval):
```
Comment (dantavori) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:683]:
```suggestion
    query = f"config timeframe = 10y | dataset = {dataset}"
    num_of_attempts = timeout // interval
    for i in range(num_of_attempts):
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 4]
```diff
@@ -602,6 +607,127 @@ def verify_test_data_exists(test_data_path: Path) -> Tuple[List[str], List[str]]
...
+            # if we don't have results from the dataset immediately we will continue to try until the timeout.
+            else:
+                logger.info(
+                    f"[cyan]trying to get results from the dataset for the {i + 1}th time. continuing to try to get the results.[/cyan]",
+                    extra={"markup": True},
+                )
+        # If the dataset doesn't exist HTTPError exception is raised.
+        # except requests.exceptions.HTTPError:
+        except XsiamApiQueryError as e:
+            if i + 1 == timeout // interval:
```
Comment (dantavori) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:703]:
```suggestion
            if i + 1 == num_of_attempts:
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 5]
```diff
@@ -828,6 +830,119 @@ def is_test_data_exists_on_server(
...
+        "[cyan]Verifying that the event IDs does not exist on the tenant[/cyan]",
+        extra={"markup": True},
+    )
+    success_msg = "[green]The event IDs does not exists on the tenant[/green]"
+    error_msg = "[red]The event id already exists in the tenant[/red]"
+    validate_expected_values_test_cases = []
+
+    for rule in modeling_rule.rules:
+        validate_event_id_does_not_exist_on_tenant_test_case = TestCase(
+            f"Validate event_id_does_not_exist_on_tenant {modeling_rule.path} dataset:{rule.dataset} "
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:855]:
```suggestion
            f"Validate event_id_does_not_exist_on_tenant {get_relative_path_to_content(modeling_rule.path)} dataset:{rule.dataset} "
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 6]
```diff
@@ -828,6 +830,119 @@ def is_test_data_exists_on_server(
...
+        test_data_event_ids = [
+            f'"{d.test_data_event_id}"'
+            for d in test_data.data
+            if d.dataset == rule.dataset
+        ]
+        td_event_ids = ", ".join(test_data_event_ids)
+        query = f"config timeframe = 10y | datamodel dataset in({rule.dataset}) | filter {rule.dataset}.test_data_event_id in({td_event_ids})"
+
+        try:
+            result = xsiam_execute_query(xsiam_client, query)
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:868]:
```suggestion
                result = retrying_caller(xsiam_execute_query, xsiam_client, query)
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 7]
```diff
@@ -828,6 +830,119 @@ def is_test_data_exists_on_server(
...
+                    success_msg
+                ]
+            else:
+                logger.error(
+                    error_msg,
+                    extra={"markup": True},
+                )
+                validate_event_id_does_not_exist_on_tenant_test_case.result += [
+                    error_msg
+                ]
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:891]:
```suggestion
                validate_event_id_does_not_exist_on_tenant_test_case.result += [Error(error_msg)
                ]
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 8]
```diff
@@ -828,6 +830,119 @@ def is_test_data_exists_on_server(
...
+                extra={"markup": True},
+            )
+            validate_event_id_does_not_exist_on_tenant_test_case.result += [success_msg]
+        else:
+            if not result:
+                logger.info(
+                    success_msg,
+                    extra={"markup": True},
+                )
+                validate_event_id_does_not_exist_on_tenant_test_case.result += [
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:881]: no need to add a success to the `.result` only errors

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 9]
```diff
@@ -828,6 +830,119 @@ def is_test_data_exists_on_server(
...
+        xsiam_client (XsiamApiClient): Xsiam API client.
+        modeling_rule (ModelingRule): Modeling rule object parsed from the modeling rule file.
+        test_data (init_test_data.TestData): Test data object parsed from the test data file.
+    """
+    logger.info(
+        "[cyan]Verifying that the event IDs does not exist on the tenant[/cyan]",
+        extra={"markup": True},
+    )
+    success_msg = "[green]The event IDs does not exists on the tenant[/green]"
+    error_msg = "[red]The event id already exists in the tenant[/red]"
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:850]: error message that's being added to TestCase data, can't contain color markers

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 10]
```diff
@@ -1325,6 +1444,12 @@ def test_modeling_rule(
         "--log-file-path",
         help="Path to the log file. Default: ./demisto_sdk_debug.log.",
     ),
+    delete_existing_dataset: bool = typer.Option(
+        False,
+        "--delete_existing_dataset",
+        "-dd",
+        help=("Deletion of the existing dataset from the tenant. Default: False."),
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:1451]:
```suggestion
        help="Deletion of the existing dataset from the tenant. Default: False.",
```

**Review by kobymeir — CHANGES_REQUESTED** [review]

**Review by kobymeir — APPROVED** [review]

**Pull request merged**